micropipenv.py

```diff
@@ -728,24 +734,26 @@ def install(
 ):  # type: (Optional[str], bool, bool, Optional[List[str]]) -> None
     """Perform installation of requirements based on the method used."""
     if method is None:
-        try:
-            _traverse_up_find_file("Pipfile.lock")
-            method = "pipenv"
-        except FileNotFound as exc:
+        paths = []
+        for file_name in _FILE_METHOD_MAP.keys():
             try:
-                _LOGGER.info("Failed to find Pipfile.lock, looking up poetry.lock: %s", str(exc))
-                _traverse_up_find_file("poetry.lock")
-                method = "poetry"
-            except FileNotFound as exc:
-                try:
-                    _LOGGER.info("Failed to find Poetry lock file, looking up requirements.txt: %s", str(exc))
-                    _traverse_up_find_file("requirements.txt")
-                    method = "requirements"
-                except FileNotFound as exc:
-                    raise FileNotFound(
-                        "Failed to find Pipfile.lock, poetry.lock or requirements.txt "
-                        "in the current directory or any of its parent: {}".format(os.getcwd())
-                    ) from exc
+                paths.append(Path(_traverse_up_find_file(file_name)))
+            except FileNotFound:
+                pass
+
+        if not paths:
+            raise FileNotFound(
+                "Failed to find Pipfile.lock, poetry.lock or requirements.txt "
+                "in the current directory or any of its parent: {}".format(os.getcwd()))
+
+        _LOGGER.debug("Dependencies definitions found: %s", str(paths))
```

Comment:
The cast to `str` should not be required to save some cycles for higher log-levels.